### PR TITLE
Docs: mpi4py in MPI-Conda Deps

### PR DESCRIPTION
**Pull request by ax3l**
Fix missing mpi4py dependency that we can get from conda.

### CODE REVIEW

File: Docs/source/install/dependencies.rst [hunk 1]
```diff
@@ -75,7 +75,7 @@ Conda (Linux/macOS/Windows)
 
       .. code-block:: bash
 
-         conda create -n warpx-cpu-mpich-dev -c conda-forge blaspp boost ccache cmake compilers git lapackpp "openpmd-api=*=mpi_mpich*" python numpy pandas scipy yt "fftw=*=mpi_mpich*" pkg-config matplotlib mamba ninja mpich pip virtualenv
+         conda create -n warpx-cpu-mpich-dev -c conda-forge blaspp boost ccache cmake compilers git lapackpp "openpmd-api=*=mpi_mpich*" python numpy pandas scipy yt "fftw=*=mpi_mpich*" pkg-config matplotlib mamba mpich mpi4py ninja pip virtualenv
```
Comment (RemiLehe) [inline Docs/source/install/dependencies.rst:78]: Thanks! 
Should we also add `mpi4py` in the `spack`, `brew` and `apt-get` instructions?

Comment (ax3l) [reply Docs/source/install/dependencies.rst:78]: I think I installed those there via pip, lemme check

Comment (ax3l) [reply Docs/source/install/dependencies.rst:78]: Spack: `py-mpi4py` is in the specs already :)

Comment (ax3l) [reply Docs/source/install/dependencies.rst:78]: Adding to Ubuntu/Debian now.

Brew has no Python at all yet... maybe another time. Tricky to get its shipped Python right sometimes.

**Review by RemiLehe — APPROVED** [review]

**Pull request merged**